renku_notebooks/util/kubernetes_.py

```diff
@@ -299,3 +301,96 @@ def _secret_exists(name, namespace):
     except client.rest.ApiException:
         pass
     return False
+
+
+def create_pvc(
+    name,
+    username,
+    git_namespace,
+    project_id,
+    project,
+    branch,
+    commit_sha,
+    git_host,
+    storage_size,
+    storage_class="default",
+):
+    """Create a PVC."""
+
+    # check if we already have this PVC
+    pvc = _get_pvc(username, project_id, commit_sha)
```

Comment:
@rokroskar I would use the `servername` / `make_servername` here. The servername we pass on to JH and that is what JH uses to distinguish between different sessions. So it is nice to reinforce and reuse that logic with the pvc rather than filtering by labels. Then when we change `make_servername` we update the logic of what distinguishes unique servers and pvcs in one go.

If you use `servername` here you dont even have to filter by labels - you can just get the exact pvc - if it exists.

Reply:
good idea - I switched to using `make_servername` for the name itself but I didn't think to just fetch by name.